packages/fluentui/react-northstar/src/components/Table/TableRow.tsx

```diff
@@ -52,94 +51,111 @@ const handleVariablesOverrides = variables => predefinedProps => ({
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(TableRow.handledProps, props);
+  const getA11yProps = useAccessibility(accessibility, {
+    debugName: TableRow.displayName,
+    actionHandlers: {
+      // https://github.com/microsoft/fluent-ui-react/issues/2150
+      unsetRowTabbable: e => {
+        rowRef.current.setAttribute('tabindex', '-1');
+      },
+      performClick: e => {
+        handleClick(e);
+      },
     },
-  };
+    mapPropsToBehavior: () => ({
+      selected,
+      header,
+    }),
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<TableRowStylesProps>(TableRow.displayName, {
+    className: tableRowClassName,
+    mapPropsToStyles: () => ({
+      header,
+      compact,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  handleClick = (e: React.SyntheticEvent) => {
+  const handleClick = (e: React.SyntheticEvent) => {
     if (e.currentTarget === e.target) {
-      _.invoke(this.props, 'onClick', e, this.props);
+      _.invoke(props, 'onClick', e, props);
       e.preventDefault();
     }
   };
 
-  renderCells(accessibility: ReactAccessibilityBehavior, variables: ComponentVariablesObject) {
-    const { items } = this.props;
-
-    const cellAccessibility = accessibility.childBehaviors ? accessibility.childBehaviors.cell : undefined;
+  const renderCells = () => {
+    // TODO: consider moving to useAccessibility()
+    const childBehavior = accessibility && accessibility({ header, selected }).childBehaviors;
 
     return _.map(items, (item: TableCellProps, index: number) => {
-      const cellProps = {
-        accessibility: cellAccessibility as Accessibility,
-      };
       const overrideProps = handleVariablesOverrides(variables);
-
       return TableCell.create(item, {
-        defaultProps: () => cellProps,
-        overrideProps,
+        defaultProps: () =>
+          getA11yProps('cell', {
+            accessibility: childBehavior ? childBehavior.cell : undefined,
+            ...overrideProps,
+          }),
       });
     });
-  }
-
-  renderComponent({
-    accessibility,
-    ElementType,
-    classes,
-    variables,
-    unhandledProps,
-  }: RenderResultConfig<any>): React.ReactNode {
-    const { children } = this.props;
-    const hasChildren = childrenExist(children);
-
-    return (
-      <Ref innerRef={this.rowRef}>
+  };
+
+  const element = (
+    <Ref innerRef={rowRef}>
+      {getA11yProps.unstable_wrapWithFocusZone(
         <ElementType
-          className={classes.root}
-          onClick={this.handleClick}
-          {...accessibility.attributes.root}
-          {...unhandledProps}
-          {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+          {...getA11yProps('root', {
+            className: classes.root,
+            onClick: handleClick,
```

Comment:
shouldn't onClick be handledProp? If it is not it will be overridden by client's onClick. Can you check this?

Reply:
Yep, you are right, we'd better add it to handledProps